base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +576,100 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMail)(LHANDLE, ULONG_PTR, lpMapiMessage, FLAGS, ULONG);
+typedef ULONG (WINAPI *FN_MAPISendMailW)(LHANDLE, ULONG_PTR, lpMapiMessageW, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszPathName)
+{
+    CStringW strFileTitle;
+    if (PathFileExistsW(pszPathName)) // File does exist?
+    {
+        strFileTitle = PathFindFileNameW(pszPathName);
+    }
+    else // Not existing
+    {
+        // Delete the temporary file if any
+        if (g_szTempFile[0])
+            ::DeleteFileW(g_szTempFile);
+
+        // Get the name of a temporary file
+        WCHAR szTempDir[MAX_PATH];
+        ::GetTempPathW(_countof(szTempDir), szTempDir);
+        if (!::GetTempFileNameW(szTempDir, L"afx", 0, g_szTempFile))
+            return MAPI_E_FAILURE;
+
+        // Save it to the temporary file
+        HBITMAP hbm = imageModel.CopyBitmap();
+        BOOL ret = ::SaveDIBToFile(hbm, g_szTempFile, FALSE);
+        ::DeleteObject(hbm);
+
+        if (!ret)
+        {
+            g_szTempFile[0] = UNICODE_NULL;
+            return MAPI_E_FAILURE; // Failure
+        }
+
+        // Use the temporary file 
+        pszPathName = g_szTempFile;
+        strFileTitle.LoadString(IDS_DEFAULTFILENAME);
+        strFileTitle += L".png";
+    }
+
+    // Load "mapi32.dll"
+    HINSTANCE hMAPI = LoadLibraryW(L"mapi32.dll");
+    if (!hMAPI)
+        return MAPI_E_NOT_SUPPORTED;
+
+    // Attachment
+    MapiFileDescW attachmentW = { 0 };
+    attachmentW.nPosition = (ULONG)-1;
+    attachmentW.lpszPathName = (LPWSTR)pszPathName;
+    attachmentW.lpszFileName = (LPWSTR)(LPCWSTR)strFileTitle;
+
+    // Message with attachment
+    MapiMessageW messageW = { 0 };
+    messageW.lpszSubject = NULL;
+    messageW.nFileCount = 1;
+    messageW.lpFiles = &attachmentW;
+
+    // First, try to open the mailer by the function of Unicode version
+    FN_MAPISendMailW pMAPISendMailW = (FN_MAPISendMailW)::GetProcAddress(hMAPI, "MAPISendMailW");
+    if (pMAPISendMailW)
+    {
+        ULONG result = pMAPISendMailW(0, (ULONG_PTR)m_hWnd, &messageW, MAPI_DIALOG | MAPI_LOGON_UI, 0);
+        ::FreeLibrary(hMAPI);
+        return result;
+    }
+
+    // Convert to ANSI strings
+    char szPathNameA[MAX_PATH];
+    char szFileTitleA[MAX_PATH];
+    ::WideCharToMultiByte(CP_ACP, 0, pszPathName, -1, szPathNameA, _countof(szPathNameA), NULL, NULL);
+    ::WideCharToMultiByte(CP_ACP, 0, strFileTitle, -1, szFileTitleA, _countof(szFileTitleA), NULL, NULL);
+
+    MapiFileDesc attachment = { 0 };
+    attachment.nPosition = (ULONG)-1;
+    attachment.lpszPathName = szPathNameA;
+    attachment.lpszFileName = szFileTitleA;
+
+    MapiMessage message = { 0 };
+    message.lpszSubject = NULL;
+    message.nFileCount = 1;
+    message.lpFiles = &attachment;
+
+    // Try again but in ANSI version
+    FN_MAPISendMail pMAPISendMail = (FN_MAPISendMail)::GetProcAddress(hMAPI, "MAPISendMail");
```

Comment:
OK with that (after reading the comments, MSDN and also https://github.com/microsoft/win32metadata/blob/81344326b62096e7c7d62bc9905d4258103eef1d/generation/WinSDK/RecompiledIdlHeaders/um/MapiUnicodeHelp.h#L8 )